react-dataflow-dynamic/src/BrushableScatterplot.js

```diff
@@ -0,0 +1,49 @@
+import {useEffect, useRef, useState} from "react";
+
+function BrushableScatterplot({height, setSelection}) {
+  const ref = useRef();
+  const [module, setModule] = useState();
+
+  useEffect(() => {
+    async function load() {
+      // Dynamically load the Runtime and the latest published version of the
+      // notebook, https://observablehq.com/@d3/brushable-scatterplot.
+      const {Runtime, Inspector} = await import("https://cdn.jsdelivr.net/npm/@observablehq/runtime@4/dist/runtime.js");
+      const {default: notebook} = await import("https://api.observablehq.com/@d3/brushable-scatterplot.js?v=3");
+
+      const runtime = new Runtime();
+      const main = runtime.module(notebook, name => {
+        // Embed the chart.
+        if (name === "viewof selection") {
+          return new Inspector(ref.current);
+        }
+        // Propagate selection state from Observable to React.
+        if (name === "selection") {
+          return {fulfilled: setSelection};
+        }
+      });
+      setModule(main);
+      return () => {
+        setModule(undefined);
+        runtime.dispose();
+      };
```

Comment:
This clean up isn't returned to useEffect because of the async…

Reply:
adopted that pattern for now ^^

"a second invocation of the effect…" — i thought the empty dependency array meant it'd only run once? i guess it doesn't guarantee that?

"ref.current could be undefined…" — so the custom observer could check if ref.current is truthy?